todoapp/src/components/Todo.vue

```diff
@@ -0,0 +1,23 @@
+<!-- eslint-disable vue/multi-word-component-names -->
+
+<script setup lang="ts">
+import type { TextIdString } from '../types/text'
+
+const props = defineProps<{
+  todos: TextIdString[]
+}>()
+const emits = defineEmits<{
+  (e: 'removeTodo', todo: TextIdString): void
```

Comment:
If you has and ID into todo object you don't have to send back the whole todo for delete action can be find and delete by id or even simple since you have an array of todo's you don't need that id and the edit and delete can be done base on the index